cmd/smoke-test/smoketest_test.go

```diff
@@ -0,0 +1,183 @@
+	var protoBuilder strings.Builder
+
+	protoBuilder.WriteString("update: <")
+	protoBuilder.WriteString(makePathProto(target, path))
+	protoBuilder.WriteString(" val: <string_val: '")
+	protoBuilder.WriteString(value)
+	protoBuilder.WriteString("'>>")
+	return protoBuilder.String()
+}
+
+func findPathValue(response *gpb.GetResponse, path string) (string, error) {
+	if len(response.Notification) != 1 {
+		return "", errors.New("response notifications must have one entry")
+	}
+
+	pathElements := utils.SplitPath(path)
+	responsePathElements := response.Notification[0].Update[0].Path.Elem
+	for pathIndex, pathElement := range pathElements {
+		responsePathElement := responsePathElements[pathIndex]
+		if pathElement != responsePathElement.Name {
+			return "", fmt.Errorf("element at %d dos not match - want %s got %s", pathIndex, pathElement, responsePathElement.Name)
+		}
+	}
+	value := response.Notification[0].Update[0].Val
+	if value == nil {
+		return "", fmt.Errorf("no value found for path %s", path)
+	}
+	return utils.StrVal(value), nil
+}
+
+func get(ctx context.Context, c client.Impl, device string, path string) (string, error) {
+	protoPath := makePathProto(device, path)
+
+	getTZRequest := &gpb.GetRequest{}
+	if err := proto.UnmarshalText(protoPath, getTZRequest); err != nil {
+		fmt.Printf("unable to parse gnmi.GetRequest from %q : %v", protoPath, err)
+		return "", err
+	}
+
+	response, err := c.(*gclient.Client).Get(ctx, getTZRequest)
+	if err != nil || response == nil {
+		return "", err
+	}
+
+	return findPathValue(response, path)
+}
+
+func set(ctx context.Context, c client.Impl, device string, path string, value string) error {
+	protoPath := makeSetPathProto(device, path, value)
+
+	setTZRequest := &gpb.SetRequest{}
+	if err := proto.UnmarshalText(protoPath, setTZRequest); err != nil {
+		return err
+	}
+
+	_, err := c.(*gclient.Client).Set(ctx, setTZRequest)
+	return err
+}
+
+func TestSmokeTestIntegration(t *testing.T) {
+	if testing.Short() || testing.CoverMode() == "count" {
+		t.Skip("Skipping smoke test")
+	}
+	q := client.Query{TLS: &tls.Config{}}
+	ctx := context.Background()
+
+	addrs := make([]string, 1)
+	addrs[0] = address
+	q.Addrs = addrs
+
+	certPool := x509.NewCertPool()
+	ca, err := ioutil.ReadFile(caCertPath)
+	assert.NilError(t, err)
+
+	ok := certPool.AppendCertsFromPEM(ca)
+	assert.Assert(t, ok, "failed to append CA certificates")
+
+	q.TLS.RootCAs = certPool
+
+	certificate, err := tls.LoadX509KeyPair(clientCrtPath, clientKeyPath)
+
+	assert.NilError(t, err, "could not load client key pair")
+
+	q.TLS.Certificates = []tls.Certificate{certificate}
+
+	c, err := gclient.New(ctx, client.Destination{
+		Addrs:       q.Addrs,
+		Target:      q.Target,
+		Timeout:     q.Timeout,
+		Credentials: q.Credentials,
+		TLS:         q.TLS,
+	})
+	assert.NilError(t, err)
+	assert.Assert(t, c != nil, "Fetching client returned nil")
+
+	// First lookup should return an error - the path has not been given a value yet
```

Comment:
my suggestion to make this modular is to have these as functions down below in the class and store them in an array in this "main" method. Iterating over the array and calling the function woudl execute the tests. anybody who wants to add his own could just writ a function and add it to that array.